components/application-mgt/org.wso2.carbon.identity.application.mgt/src/main/java/org/wso2/carbon/identity/application/mgt/dao/impl/ApplicationDAOImpl.java

```diff
@@ -2162,6 +2181,66 @@ public ApplicationBasicInfo[] getApplicationBasicInfo(String filter, int offset,
         return appInfo.toArray(new ApplicationBasicInfo[0]);
     }
 
+    /**
+     * Get query to exclude system portals if excludeSystemPortals is true.
+     *
+     * @param excludeSystemPortals  Exclude system portals.
+     * @param useTableNameInQuery   Use the table name in query.
+     * @return String Query to exclude system portals.
+     */
+    private String populateSystemPortalsExcludeQuery(Boolean excludeSystemPortals, Boolean useTableNameInQuery) {
+
+        if (excludeSystemPortals) {
+            String excludeSystemPortalsQuery = useTableNameInQuery ?
+                    ApplicationMgtDBQueries.EXCLUDE_SYSTEM_PORTALS_BY_TABLE_NAME_AND_NAME :
+                    ApplicationMgtDBQueries.EXCLUDE_SYSTEM_PORTALS_BY_NAME;
+            Set<String> systemPortals = getSystemPortalsFromConfiguration(SYSTEM_PORTALS);
+            return systemPortals.isEmpty() ? "" :
+                    String.format(excludeSystemPortalsQuery, systemPortals.stream()
+                                    .map(s -> "'" + s + "'")
+                                    .collect(Collectors.joining(", ")));
+        } else {
+            return "";
```

Comment:
Lets update other places as well.
```suggestion
            return StringUtils.EMPTY;
```